### PR TITLE
Fix multiple potential vulnerabilities and bugs

### PR DESCRIPTION
**Pull request by YangY-Xiao**
Fix multiple potential vulnerabilities and bugs

### CODE REVIEW

File: src/bin/jpwl/convert.c [hunk 1]
```diff
@@ -135,15 +131,19 @@ static int tga_readheader(FILE *fp, unsigned int *bits_per_pixel,
     id_len = (unsigned char)tga[0];
     cmap_type = (unsigned char)tga[1];
     image_type = (unsigned char)tga[2];
-    cmap_index = get_ushort(*(unsigned short*)(&tga[3]));
+    cmap_index = get_tga_ushort(*(unsigned short*)(&tga[3]));
     cmap_len = get_ushort(*(unsigned short*)(&tga[5]));
+    cmap_len = get_tga_ushort(*(unsigned short*)(&tga[5]));
```
Comment (rouault) [inline src/bin/jpwl/convert.c]: I suspect this doesn't compile... Line 135 should be removed as it is effectively overrideen by line 136. Similar instances below

Comment (Young-X) [reply src/bin/jpwl/convert.c]: I have fixed that by make a new pr.

**Comment by Young-X** [comment]
Ping

**Comment by ret2libc** [comment]
> @Young-X I had to revert that commit in e1740e7 since it didn't compile. I'm disappointed you submit commits that don't even compile...

@rouault it seems commit https://github.com/uclouvain/openjpeg/commit/e1740e7ce79d0a1676db4da0f4189b64e85f52cb was supposed to fix CVE-2018-20846, which is about out-of-bound accesses in pi_next_lrcp, pi_next_rlcp, pi_next_rpcl, pi_next_pcrl, pi_next_rpcl, and pi_next_cprl . With that commit reverted, is there a patch for that CVE? Thanks.

**Comment by rouault** [comment]
> is there a patch for that CVE?

no

**Pull request merged**